@philipphack the CI failures look legit, can you PTAL?  E.g. on Ubuntu CPU I see:

```
tensorflow/core/common_runtime/eager/eager_op_rewrite_registry_test.cc:42:75: error: macro "REGISTER_REWRITE" requires 3 arguments, but only 2 given
 REGISTER_REWRITE(EagerOpRewriteRegistry::PRE_EXECUTION, TestEagerOpRewrite);
                                                                           ^
tensorflow/core/common_runtime/eager/eager_op_rewrite_registry_test.cc:42:1: error: 'REGISTER_REWRITE' does not name a type
 REGISTER_REWRITE(EagerOpRewriteRegistry::PRE_EXECUTION, TestEagerOpRewrite);
 ^~~~~~~~~~~~~~~~
```

--- a/tensorflow/core/common_runtime/eager/eager_op_rewrite_registry.h
+++ b/tensorflow/core/common_runtime/eager/eager_op_rewrite_registry.h
@@ -57,7 +57,6 @@ class EagerOpRewriteRegistry {
   };
 
   // Add a rewrite pass to the registry.
-  // Only one rewrite pass is allowed per phase.
   void Register(Phase phase, int32 ordinal,
                 std::unique_ptr<EagerOpRewrite> pass);
 

--- a/tensorflow/core/common_runtime/eager/eager_op_rewrite_registry_test.cc
+++ b/tensorflow/core/common_runtime/eager/eager_op_rewrite_registry_test.cc
@@ -39,7 +39,11 @@ class TestEagerOpRewrite : public EagerOpRewrite {
 
 int TestEagerOpRewrite::count_ = 0;
 
-REGISTER_REWRITE(EagerOpRewriteRegistry::PRE_EXECUTION, TestEagerOpRewrite);
+// Register two rewriter passes during the PRE_EXECUTION phase
+REGISTER_REWRITE(EagerOpRewriteRegistry::PRE_EXECUTION, 10000,
+                 TestEagerOpRewrite);
+REGISTER_REWRITE(EagerOpRewriteRegistry::PRE_EXECUTION, 10001,
+                 TestEagerOpRewrite);
 
 TEST(EagerOpRewriteRegistryTest, RegisterRewritePass) {
   EXPECT_EQ(0, TestEagerOpRewrite::count_);
@@ -54,7 +58,7 @@ TEST(EagerOpRewriteRegistryTest, RegisterRewritePass) {
   EXPECT_EQ(Status::OK(),
             EagerOpRewriteRegistry::Global()->RunRewrite(
                 EagerOpRewriteRegistry::PRE_EXECUTION, &orig_op, &out_op));
-  EXPECT_EQ(1, TestEagerOpRewrite::count_);
+  EXPECT_EQ(2, TestEagerOpRewrite::count_);
   EXPECT_EQ("NoOp", out_op->Name());
   ctx->Unref();
 }

